server/src/main/java/org/elasticsearch/cluster/metadata/RollupGroup.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.cluster.metadata;
+
+import org.elasticsearch.cluster.AbstractDiffable;
+import org.elasticsearch.cluster.Diff;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Object representing information about rollup-v2 indices and their respective original-indexes. These objects
+ * also include information about their capabilities, like which date-intervals and date-timezones they are configured
+ * with. Used by {@link RollupMetadata}.
+ */
+public class RollupGroup extends AbstractDiffable<RollupGroup> implements ToXContentObject {
+    private static final ParseField GROUP_FIELD = new ParseField("group");
+    private static final ParseField DATE_INTERVAL_FIELD = new ParseField("interval");
+    private static final ParseField DATE_TIMEZONE_FIELD = new ParseField("timezone");
+
+    // the list of indices part of this rollup group
+    private List<String> group;
+    // a map from index-name to the date interval used in the associated index
+    private Map<String, String> dateInterval;
+    // a map from index-name to timezone used in the associated index
+    private Map<String, String> dateTimezone;
```

Comment:
I can wrap ZoneId for sure. ZoneId itself cannot be used because it is not Writeable

Reply:
Yeah, I was thinking for the internal usage, not for the serialization/deserialization layer

Reply:
got it, I'd prefer if it were a simple object that serialized and was not a concern of this class. So I'll try making a thin Writeable wrapper for ZoneId and others. Let me know if you think that is overkill and the String->Obj conversion should occur within this class to reduce extra class hierarchies. I see both sides, but lean towards wrapping.

Reply:
Making it not this class's problem seems sensible.  What I want is for it not to be the problem of clients of this class either.  I think a wrapper solves both, and actually I'm a little surprised we haven't needed one already.  I think that's got my vote.  I'd also be okay with leaving it as a ToDo for now, since it doesn't look like anything is reading these values yet?